core/src/main/scala/org/apache/spark/util/CommandLineUtils.scala

```diff
@@ -30,7 +30,11 @@ private[spark] trait CommandLineUtils extends CommandLineLoggingUtils {
 
 private[spark] trait CommandLineLoggingUtils {
   // Exposed for testing
-  private[spark] var exitFn: Int => Unit = (exitCode: Int) => System.exit(exitCode)
+  private[spark] var exitFn: (Int, Option[Throwable]) => Unit =
+    (exitCode: Int, cause: Option[Throwable]) => {
+      cause.foreach(_.printStackTrace(printStream))
```

Comment:
This is the actual improvement, isn't it, @pan3793 ?

Although I got the idea, I don't think this contribution is relevant to SPARK-53170 and SPARK-48547. Those are technically in the same domain, but the previous goal was not having a user-friendly log in `printStream`.

Reply:
Yes, exactly.